File: 1814-count-nice-pairs-in-an-array/1814-count-nice-pairs-in-an-array.java
class Solution {
public:
    int countNicePairs(vector<int>& nums) {
        vector<int> arr;
        for(int i=0; i<nums.size(); i++) arr.push_back(nums[i]-rev(nums[i]));
        unordered_map<int, int> mp;
        int ans = 0, mod = 1e9+7;
        for(int x : arr){
            ans = (ans + mp[x]) % mod;
            mp[x]++;
        }
        return ans;
    }
    int rev(int num){
        int res = 0;
        while(num > 0){
            res = res * 10 + num%10;
            num /= 10;
        }
        return res;
    }
};

File: 1814-count-nice-pairs-in-an-array/1814-count-nice-pairs-in-an-array.py
class Solution:
    def countNicePairs(self, nums: List[int]) -> int:
        def rev(num):
            res = 0
            while num:
                res = res * 10 + num % 10
                num //= 10
            return res
        arr = []
        for i in range(len(nums)):
            arr.append(nums[i]-rev(nums[i]))
        mp = defaultdict(int)
        ans, mod = 0, 10**9+7
        for x in arr:
            ans = (ans + mp[x]) % mod
            mp[x] += 1
        return ans